test/Events.t.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+
+import { MockERC20, PSMTestBase } from "test/PSMTestBase.sol";
+
+contract PSMEventTests is PSMTestBase {
+
+    event Swap(
+        address indexed assetIn,
+        address indexed assetOut,
+        address sender,
+        address indexed receiver,
+        uint256 amountIn,
+        uint256 amountOut,
+        uint16  referralCode
+    );
+
+    event Deposit(
+        address indexed asset,
+        address indexed user,
+        uint256 assetsDeposited,
+        uint256 sharesMinted,
+        uint16  referralCode
+    );
+
+    event Withdraw(
+        address indexed asset,
+        address indexed user,
+        uint256 assetsWithdrawn,
+        uint256 sharesBurned,
+        uint16  referralCode
+    );
+
+    address sender   = makeAddr("sender");
+    address receiver = makeAddr("receiver");
+
+    function test_deposit_events() public {
+        vm.startPrank(sender);
+
+        dai.mint(sender, 100e18);
+        dai.approve(address(psm), 100e18);
+
+        vm.expectEmit();
```

Comment:
The first argument is of the contract emitter. It's best practise to set this for robustness. IE `vm.expectEmit(address(psm));`. Do this for all instances.

Reply:
Nice updated